File: frontend/src/types/book.ts
export interface Book {
  id: string;
  title: string;
  author: string;
  isbn: string;
  coverImageUrl: string | null;
  rating: number | null;
  comments: string | null;
}

export interface BookListResponse {
  items: Book[];
  totalCount: number;
}

export type SortOrder = 'asc' | 'desc';

export interface GetBooksParams {
  page: number;
  pageSize: number;
  search?: string;
  sort: SortOrder;
}

export interface CreateBookPayload {
  title: string;
  author: string;
  isbn: string;
  coverImageUrl: string;
  rating: number | null;
  comments: string | null;
}

export interface UpdateBookPayload {
  rating: number | null;
  comments: string | null;
}

File: backend/src/Books.Api/Models/Book.cs
namespace Books.Api.Models;

public sealed class Book
{
    public Guid Id { get; init; } = Guid.NewGuid();
    public string Title { get; set; } = string.Empty;
    public string Author { get; set; } = string.Empty;
    public string Isbn { get; set; } = string.Empty;
    public string? CoverImageUrl { get; set; }
    public int? Rating { get; set; }
    public string? Comments { get; set; }
}
